pkg/serverless/serverless.go

```diff
@@ -256,13 +255,18 @@ func WaitForNextInvocation(stopCh chan struct{}, statsdServer *dogstatsd.Server,
 		return fmt.Errorf("WaitForNextInvocation: can't unmarshal the payload: %v", err)
 	}
 
-	// sets the current ARN.
-	// TODO(remy): we could probably do this once
-	if payload.InvokedFunctionArn != "" {
+	if payload.EventType == "INVOKE" {
+		log.Debug("Received invocation event")
 		aws.SetARN(payload.InvokedFunctionArn)
 	}
-
 	if payload.EventType == "SHUTDOWN" {
+		log.Debug("Received shutdown event. Reason: " + payload.ShutdownReason)
+		if payload.ShutdownReason == "timeout" {
+			metricTags := getTagsForEnhancedMetrics()
+			sendTimeoutEnhancedMetric(metricTags, metricsChan)
+			// Ensure that timeout metric has been received by the statsdServer before flushing
+			time.Sleep(50 * time.Millisecond)
```

Comment:
I know we need an arbitary delay here, but how did we get 50ms, and can we move this out into a constant?

Reply:
It's arbitrary, I just picked a smallish number, I can test with some smaller numbers and see what happens.